### PR TITLE
cart page

### PR DESCRIPTION
**Pull request by harshalini**
Added cart page in vivir fashion ecommerce site.
Hover on the card and you will see the add to cart option.
Now, a product can be added to cart and removed from the cart.
Quantity of selected product can be changed &
Price of all products in the cart is calculated accordingly.
Check the preview here ->
https://623cca8d1f9be832fc9a7323--chipper-medovik-91fa6e.netlify.app/

### CODE REVIEW

File: src/components/cart-components/cart-item.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { useCart } from "../../context/cart-context";
+export const CartItem = () => {
+    const { cartState: { cart }, cartDispatch } = useCart()
+    return (
+        <div>
+            {cart.map((prod) => {
+                return (
+                    <div className="cart-product"
+                        key={prod.id}
+                        style={{ marginBottom: "2.5rem" }}>
```
Comment (sushil-mod) [inline src/components/cart-components/cart-item.js:10]: if possible remove inline style

File: src/components/cart-components/table-bar.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export const TableBar = () => {
+    return (
+        <div className="cart-product" style={{ marginTop: "6rem" }}>
```
Comment (sushil-mod) [inline src/components/cart-components/table-bar.js]: same here inline style

File: src/components/cart-components/cart-summery.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { useCart } from "../../context/cart-context";
+import { useState, useEffect } from "react";
+export const CartSummery = () => {
+    const { cartState: { cart } } = useCart();
+    const [total, setTotal] = useState();
+
+    useEffect(() => {
+        setTotal(cart.reduce((acc, curr) => acc + curr.discountedPrice * curr.qty, 0));
+    }, [cart]);
```
Comment (JantuDeb) [inline src/components/cart-components/cart-summery.js]: no need to use `useEffect`
```
const total = cart.reduce((acc, curr) => acc + curr.discountedPrice * curr.qty, 0)
```

**Review by JantuDeb — APPROVED** [review]
left a comment

File: src/components/products/product-card.js [hunk 4]
```diff
@@ -1,6 +1,9 @@
 import { useProduct } from "../../context/productList-context";
 import { SortedPrice, ProductRatings, GetFliteredProducts } from "../../filter-utils/filterUtils";
+import { useCart } from "../../context/cart-context";
+import { Link } from "react-router-dom";
 export function GetProducts() {
+  const { cartState: {cart}, cartDispatch } = useCart()
   const { product } = useProduct();
   const getSortedProducts = SortedPrice(product)
   const getRatingProducts = ProductRatings(getSortedProducts)
```
Comment (sushil-mod) [inline src/components/products/product-card.js]: your code is also correct , if possible you can try this with compose function ( if i am wrong please correct me )

const compose = ( ...getfunction ) =>( state , product ) =>getfunction.reduce( ( data , getfunction  ) =>getfunction( state , data ) , product );

const filteredProduct = compose(  SortedPrice , ProductRating , GetFilteredProducts   ) ( state , product  ) 

filteredProduct will contain all filtered products

**Review by sushil-mod — APPROVED** [review]
just a suggestion if you have time you can implement , overall  code is nice

File: src/components/products/product-card.js (hunk 4 above)
Comment (harshalini) [reply src/components/products/product-card.js]: Thank you for the suggestion! It is correct but it is also working fine without the state.

File: src/components/cart-components/cart-summery.js (hunk 3 above)
Comment (harshalini) [reply src/components/cart-components/cart-summery.js]: Ok, thank you for pointing out that!

**Pull request merged**